testcontainers/src/clients/cli.rs

```diff
@@ -241,35 +247,25 @@ impl Client {
     }
 }
 
-impl Default for Cli {
-    fn default() -> Self {
-        Self::new::<env::Os>()
-    }
-}
-
-impl Cli {
+impl Client {
     pub fn new<E>() -> Self
     where
         E: GetEnvValue,
     {
         Self {
-            inner: Arc::new(Client {
-                container_startup_timestamps: Default::default(),
-                created_networks: Default::default(),
-                binary: "docker".into(),
-                command: env::command::<E>().unwrap_or_default(),
-            }),
+            container_startup_timestamps: Default::default(),
+            created_networks: Default::default(),
+            binary: "docker".into(),
+            command: env::command::<E>().unwrap_or_default(),
         }
     }
 }
 
-impl Docker for Cli {
+impl Docker for &Client {
```

Comment:
Do we still need the `Docker` trait now? By virtue of the `RunViaCli` trait, the implementation is no longer pluggable so I believe we could just remove it now.